### ISSUE TITLE
Style Visualization

### ISSUE THREAD

**Issue by YuchenLiu98** [HeliosZhao/SHADE#3]
Congratulations on this interesting and valuable work.
Could you provide the code for visualition of stylized images?
"To better understand our SHM, we visualize the style-diversified samples with an autoencoder"

Thanks a lot for your help.

**Comment by HeliosZhao** [comment]
Thanks for your interest in our work. The implementation of visualization is as follows.

We take the fixed pre-trained blocks before SHM as the encoder, and add an additional symmetrical decoder. The decoder is trained by L1 loss with the source data. After training, we use SHM to generate multiple styles and replace the style of the original sample. Then we use the decoder to generate the visualization results.

**Comment by YuchenLiu98** [comment]
Thanks for your reply. So "fixed pre-trained blocks before SHM" is imagenet pretrained and the decoder is exactly the opposite of the encoder?

**Comment by HeliosZhao** [comment]
> ``fixed pre-trained blocks before SHM''

It depends on what u want to visualize. You can use the ImageNet pre-trained model or the training model at any stage. For our visualization results, we use the final model after training. 

>``the decoder is exactly the opposite of the encoder''

Here we use two deconv layers with stride 2 and kernel size 3x3. You can modify the structure as long as you can get a clear result. Note that we do not focus on image reconstruction, but just use this for better visualization.

**Comment by YuchenLiu98** [comment]
Thanks for your reply!
